enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,562 @@
+of the problems with the existing implementation.
+
+## Motivation
+
+There are a few problems with configure-ovs.sh:
+
+* It is implemented in Bash, which makes it fragile and difficult to test.
+* It is guessing at what the deployer wants for their bridge configuration,
+  and it is awkward to override when it guesses wrong.
+* It has customer-specific logic because there is no alternative for us to
+  support advanced use cases.
+* It is incompatible with day 2 changes to the bridge using things like the
+  Kubernetes-NMState operator because the operator and script configurations
+  will overwrite each other.
+
+### User Stories
+
+* As an OpenShift administrator, I want the OVNKubernetes bridge on a different
+  interface from the node IP.
+
+* As an OpenShift administrator, I want to be able to make changes to the
+  bridge interface using the same tools I do on the other interfaces.
+
+* As an OpenShift developer, I want a mechanism for bridge configuration that
+  is well-tested and less likely to break than a Bash script.
+
+* As an OpenShift developer, I want to simplify the configure-ovs.sh script
+  by providing a better way to do advanced bridge configuration.
+
+### Goals
+
+Once this design is implemented, deployers of OpenShift will be able to
+explicitly configure br-ex to their exact specifications and will be able to
+modify it after deployment using standard networking tools.
+
+### Non-Goals
+
+Complete replacement of configure-ovs. This mechanism is intended for more
+advanced use cases, but it will require more complex configuration than
+basic deployments need. A possible followup would be to reimplement
+configure-ovs.sh in terms of this new mechanism so there is a simple path
+here too, but that is not required for the initial implementation.
+
+An example of a use case this is not intended to address would be a
+single-nic deployment on a cloud platform with no unusual network settings.
+In that case manual configuration of br-ex would be an unnecessary extra
+burden on the user and the logic to handle it automatically would be fairly
+simple so using configure-ovs is far less likely to cause problems.
+
+## Proposal
+
+This mechanism will be similar to the day-1 network configuration described
+in enhancements/network/baremetal-ipi-network-configuration.md. However,
+there will be some important differences, which will be discussed below.
+
+At a high level, we will provide an interface for the deployer to specify
+per-host interface configuration. Configuration for an OpenVSwitch bridge
+named br-ex will be mandatory. The format for the configuration will be
+NMState YAML, which will be provided to each node via Ignition. Around the
+same point in the boot process that configure-ovs would run, we will run
+NMState to apply the provided configuration. This will only be done on first
+boot because subsequent changes will be made using the Kubernetes-NMState
+operator.
```

Comment:
Yep, will do.

Right now it would need to be done manually. There isn't really a way to convert it at install time because kubernetes-nmstate won't be installed yet. I suppose there may be some way to auto-create some manifests that could be applied later though.